app/views/AnalyticalFramework/utils.ts

```diff
@@ -0,0 +1,183 @@
+import { randomString } from '@togglecorp/fujs';
+import {
+    Widget,
+    ScaleWidget,
+    SingleSelectWidget,
+    MultiSelectWidget,
+    OrganigramWidget,
+    OrganigramDatum,
+    Matrix1dWidget,
+    Matrix2dWidget,
+} from '#types/newAnalyticalFramework';
+import { PartialWidget } from '#components/framework/AttributeInput';
+
+function clone(widget: PartialWidget) {
+    return ({
+        ...widget,
+        title: `${widget.title}-cloned`,
+        key: randomString(),
+        clientId: randomString(),
```

Comment:
These two may need to have the same value.
@thenav56 what do you think?

Reply:
Are we now using key instead of clientId? Or both?

Not sure where this is used. But if we have both, then we can easily remove clientId in the future (Currently server uses clientId for widget properties data)

Reply:
I have started work on PR to convert all "persistent" keys to be named "key".

One problem is that we have "id", "clientId" and "key" on widget level.
We could do away with "key", but don't know how its used on the server